soda_machine.rb

```diff
@@ -1,18 +1,33 @@
 class SodaMachine
-  attr_reader :sodas, :cash
+  attr_reader :sodas, :cash, :sold_soda
 
   def initialize(args = {})
     @sodas = args[:sodas]
     @cash = args[:cash]
+    @sold_soda = []
   end
 
   def current_inventory_count
+      @sodas.length
   end
 
   def find_soda(soda_brand)
+    if @sodas[:sodas].include? soda_brand
```

Comment:
It looks like you're thinking of `@sodas` as a Hash (maybe the hash that was passed in to initialize?).  It's not; it's an array of `Soda` objects